vars/checksPublishResults.groovy

```diff
@@ -0,0 +1,151 @@
+        aggregateReports(configuration.get('aggregation'))
+    }
+}
+
+def aggregateReports(settings){
+    if (settings.active) {
+        def options = createCommonOptionsMap('AnalysisPublisher', settings)
+        // publish
+        step(options)
+    }
+}
+
+def reportTasks(settings){
+    if (settings.active) {
+        def options = createCommonOptionsMap('TasksPublisher', settings)
+        options.put('pattern', settings.get('pattern'))
+        options.put('high', settings.get('high'))
+        options.put('normal', settings.get('normal'))
+        options.put('low', settings.get('low'))
+        // publish
+        step(options)
+    }
+}
+
+def report(publisherName, settings, doArchive){
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+        def options = createCommonOptionsMap(publisherName, settings)
+        options.put('pattern', pattern)
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive'), pattern, true)
+    }
+}
+
+def reportWarnings(parserName, settings, doArchive){
+    if (settings.active) {
+        def pattern = settings.get('pattern')
+        def options = createCommonOptionsMap('WarningsPublisher', settings)
+        options.put('parserConfigurations', [[
+            parserName: parserName,
+            pattern: pattern
+        ]])
+        // publish
+        step(options)
+        // archive check results
+        archiveResults(doArchive && settings.get('archive'), pattern, true)
+    }
+}
+
+def archiveResults(archive, pattern, allowEmpty){
+    if(archive){
+        echo "[${STEP_NAME}] archive ${pattern}"
+        archiveArtifacts artifacts: pattern, allowEmptyArchive: allowEmpty
+    }
+}
+
+@NonCPS
+def createCommonOptionsMap(publisherName, settings){
+    Map result = [:]
+    def thresholds = settings.get('thresholds', [:])
+    def fail = thresholds.get('fail', [:])
+    def unstable = thresholds.get('unstable', [:])
+
+    result.put('$class', publisherName)
+    result.put('healthy', settings.get('healthy'))
+    result.put('unHealthy', settings.get('unHealthy'))
+    result.put('canRunOnFailed', true)
+    result.put('failedTotalAll', fail.get('all'))
+    result.put('failedTotalHigh', fail.get('high'))
+    result.put('failedTotalNormal', fail.get('normal'))
+    result.put('failedTotalLow', fail.get('low'))
+    result.put('unstableTotalAll', unstable.get('all'))
+    result.put('unstableTotalHigh', unstable.get('high'))
+    result.put('unstableTotalNormal', unstable.get('normal'))
+    result.put('unstableTotalLow', unstable.get('low'))
+    // filter empty values
+    result = result.findAll {
+        return it.value != null && it.value != ''
+    }
+    return result
+}
+
+@NonCPS
+def prepare(parameters){
+    // ensure tool maps are initialized correctly
+    for(String tool : ['aggregation','tasks','pmd','cpd','findbugs','checkstyle','eslint','pylint']){
```

Comment:
What about having this list only once? We have it already above (... with "archive", which could be added above as needed). This would increase the maintainability in case we have to add new targets.

Reply:
Agreed, but I suggest to reuse the list from above and remove the `archive` entry.